### PR TITLE
 add advanced filters via properties

### PR DESCRIPTION
**Pull request by svenwltr**
@rebuy-de/prp-aws-nuke Please take a look. What do you think?

/cc @tomvachon

### CODE REVIEW

**Comment by tomvachon** [comment]
Can’t look for 4-5 days. I’ll get back to you

File: README.md [hunk 1]
```diff
@@ -313,6 +313,29 @@ Any resource whose resource identifier exactly matches any of the filters in
...
+#### Filter Properties
+
+Some resources support filtering via properties. When a resource support these
+properties, they will be listed in the output like in this example:
+
+```
+global - IAMUserPolicyAttachment - 'admin -> AdministratorAccess' - [RoleName: "admin", PolicyArn: "arn:aws:iam::aws:policy/AdministratorAccess", PolicyName: "AdministratorAccess"] - would remove
+```
+
+To use properties, it is required to specify a object with `propeties` and
```
Comment (bjoernhaeuser) [inline README.md]: `properties`

File: cmd/queue.go [hunk 2]
```diff
@@ -49,6 +52,20 @@ func (i *Item) List() ([]resources.Resource, error) {
...
 }
 
+func (i *Item) GetProperty(key string) string {
+	if key == "" {
+		return i.Resource.String()
+	}
+
+	getter, ok := i.Resource.(resources.ResourcePropertyGetter)
+	if !ok {
+		logrus.Warnf("%T does not support custom properties", i.Resource)
```
Comment (bjoernhaeuser) [inline cmd/queue.go]: IMHO we should fail if the user provided config is incorrect

Comment (svenwltr) [reply cmd/queue.go]: In this case I have to see how to bubble this error up, since we are in a goroutine. The only way that currently comes into my mind is an ugly error channel.

Comment (svenwltr) [reply cmd/queue.go]: Will do in a separate PR to keep the history clean: https://github.com/rebuy-de/aws-nuke/pull/200

**Comment by tomvachon** [comment]
As written, its hard to follow.  I dont see any major concerns tho

**Comment by svenwltr** [comment]
I needed to do some additional changes to avoid nil pointer dereferences. See 6377b788d7a709518a5e756cda2b9685f2c371c2.
